### PR TITLE
Add answer format for handling a multiple component picker.

### PR DESCRIPTION
**Pull request by syoung-smallwisdom**
Added `ORKMultipleValuePickerAnswerFormat` to handle displaying a multiple-component picker wheel. This design uses an input array of `ORKValuePickerAnswerFormat` to define the choices for each component.

simulator screen shot jan 20 2017 3 58 11 pm

### CODE REVIEW

File: ResearchKit/Common/ORKAnswerFormat.h [hunk 1]
```diff
@@ -635,6 +639,94 @@ ORK_CLASS_AVAILABLE
...
+/**
+ An array of text choices that represent the options to display in the picker. (read-only)
+ 
+ Note that the `detailText` property of each choice is ignored. Be sure to create localized text for
+ each choice that is short enough to fit in a `UIPickerView` object.
+ */
+@property (copy, readonly) NSArray<ORKTextChoice *> *textChoices;
+
+@end
+
```
Comment (umerkhan-apple) [inline ResearchKit/Common/ORKAnswerFormat.h]: Add extra new line.

File: ResearchKit/Common/ORKAnswerFormat.m [hunk 2]
```diff
@@ -610,6 +638,119 @@ - (NSString *)stringForAnswer:(id)answer {
 
 @end
```
Comment (umerkhan-apple) [inline ResearchKit/Common/ORKAnswerFormat.m]: Add extra new line

File: ResearchKit/Common/ORKMultipleValuePicker.h [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+ DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+ SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+ OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+@import UIKit;
+#import "ORKPicker.h"
+
```
Comment (umerkhan-apple) [inline ResearchKit/Common/ORKMultipleValuePicker.h]: 2 new lines before and after imports.

File: ResearchKit/Common/ORKMultipleValuePicker.m [hunk 4]
```diff
@@ -0,0 +1,255 @@
...
+ ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE
+ FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+ DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+ SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+ OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#import "ORKMultipleValuePicker.h"
```
Comment (umerkhan-apple) [inline ResearchKit/Common/ORKMultipleValuePicker.m]: 2 new lines before imports

File: ResearchKit/Common/ORKMultipleValuePicker.m [hunk 5]
```diff
@@ -0,0 +1,255 @@
...
+
+#import "ORKAccessibilityFunctions.h"
+
+
+@interface ORKMultipleValuePicker () <UIPickerViewDataSource, UIPickerViewDelegate>
+
+@property (nonatomic, strong) NSArray <ORKChoiceAnswerFormatHelper *> *helpers;
+
+@end
+
```
Comment (umerkhan-apple) [inline ResearchKit/Common/ORKMultipleValuePicker.m]: Add extra new line.

File: ResearchKit/Common/ORKMultipleValuePicker.m [hunk 6]
```diff
@@ -0,0 +1,255 @@
...
+            *stop = YES;
+        }
+    }];
+
+    _answer = (answers.count == self.helpers.count) ? answers : ORKNullAnswerValue();
+    if ([self.pickerDelegate respondsToSelector:@selector(picker:answerDidChangeTo:)]) {
+        [self.pickerDelegate picker:self answerDidChangeTo:_answer];
+    }
+}
+
```
Comment (umerkhan-apple) [inline ResearchKit/Common/ORKMultipleValuePicker.m]: Add new line

File: ResearchKit/Common/ORKMultipleValuePicker.m [hunk 7]
```diff
@@ -0,0 +1,255 @@
...
+    if (UIAccessibilityIsVoiceOverRunning()) {
+        ORKAccessibilityPerformBlockAfterDelay(0.75, ^{
+            NSArray *axElements = [self.pickerView accessibilityElements];
+            if ([axElements count] > 0) {
+                UIAccessibilityPostNotification(UIAccessibilityLayoutChangedNotification, [axElements objectAtIndex:0]);
+            }
+        });
+    }
+}
+
```
Comment (umerkhan-apple) [inline ResearchKit/Common/ORKMultipleValuePicker.m]: Add new line

File: ResearchKit/Common/ORKResult.m [hunk 8]
```diff
@@ -1452,6 +1452,62 @@ - (id)answer {
 
 @end
```
Comment (umerkhan-apple) [inline ResearchKit/Common/ORKResult.m]: add new line

File: ResearchKit/Common/ORKResult.m [hunk 9]
```diff
@@ -1452,6 +1452,62 @@ - (id)answer {
...
+    [super encodeWithCoder:aCoder];
+    ORK_ENCODE_OBJ(aCoder, componentsAnswer);
+    ORK_ENCODE_OBJ(aCoder, separator);
+}
+
+- (instancetype)initWithCoder:(NSCoder *)aDecoder {
+    self = [super initWithCoder:aDecoder];
+    if (self) {
+        ORK_DECODE_OBJ_ARRAY(aDecoder, componentsAnswer, NSObject);
+        ORK_DECODE_OBJ(aDecoder, separator);
```
Comment (umerkhan-apple) [inline ResearchKit/Common/ORKResult.m]: ORK_DECODE_OBJ_CLASS(aDecoder, separator, NSString)

**Comment by syoung-smallwisdom** [comment]
@umerkhan-apple 
Formatting and encoding changes have been addressed. Thanks!

File: ResearchKit/Common/ORKResult.m (hunk 8 above)
Comment (syoung-smallwisdom) [inline ResearchKit/Common/ORKResult.m]: done

File: ResearchKit/Common/ORKMultipleValuePicker.m (hunk 7 above)
Comment (syoung-smallwisdom) [inline ResearchKit/Common/ORKMultipleValuePicker.m]: done

File: ResearchKit/Common/ORKMultipleValuePicker.m (hunk 6 above)
Comment (syoung-smallwisdom) [inline ResearchKit/Common/ORKMultipleValuePicker.m]: done

File: ResearchKit/Common/ORKMultipleValuePicker.m (hunk 5 above)
Comment (syoung-smallwisdom) [inline ResearchKit/Common/ORKMultipleValuePicker.m]: done

File: ResearchKit/Common/ORKAnswerFormat.h (hunk 1 above)
Comment (syoung-smallwisdom) [inline ResearchKit/Common/ORKAnswerFormat.h]: done

**Comment by umerkhan-apple** [comment]
@syoung-smallwisdom Thanks for updating the PR! There are some merge conflicts that need to be resolved. Apart from that, the PR is ready to be merged! 👍

**Comment by syoung-smallwisdom** [comment]
@umerkhan-apple merge conflicts resolved. Thanks.

**Comment by umerkhan-apple** [comment]
@syoung-smallwisdom Great, thank you! Great contribution! 👍 🎉

**Pull request merged**